apollo-router/src/plugins/telemetry/config_new/spans.rs

```diff
@@ -31,7 +31,6 @@ use crate::tracer::TraceId;
 #[derive(Deserialize, JsonSchema, Clone, Default, Debug)]
 #[serde(deny_unknown_fields, default)]
 pub(crate) struct Spans {
-    #[serde(skip)]
     /// Whether to create a `request` span. This will be removed in future, and users should set this to false.
     pub(crate) legacy_request_span: bool,
```

Comment:
should this be set to true by default to avoid a breaking change?

Reply:
I think I discussed previously about this with @BrynCooke but I don't remember what was the outcome

Reply:
We are between a rock and hard place here.

We don't want new users to rely on the old span. We don't want to break existing users.

Suggest that for now it is true, and we log a warning if the user has not set it to false.
At some point in the future we flip the default value.

How does this sound?

Reply:
Ok we have now a configuration called `mode` which can be set to `legacy` or `new` and should be fully compliant with what we had previously. That's a better approach. And by default we set the `new` mode. Which we should document in the router changelog